packages/core/src/oot/actors/Custom/Custom_Torch2.c

```diff
@@ -0,0 +1,386 @@
+    gsDPLoadTextureBlock(gElegyShellHumanBootsTex, G_IM_FMT_RGBA, G_IM_SIZ_16b, 16, 16, 0, G_TX_MIRROR | G_TX_WRAP,
+                         G_TX_MIRROR | G_TX_CLAMP, 4, 4, G_TX_NOLOD, G_TX_NOLOD),
+    gsDPSetPrimColor(0, 0x80, 255, 255, 255, 255),
+    gsSPVertex(&sElegyShellHumanVtx[283], 32, 0),
+    gsSP2Triangles(0, 1, 2, 0, 3, 4, 5, 0),
+    gsSP2Triangles(6, 7, 8, 0, 6, 9, 7, 0),
+    gsSP2Triangles(10, 11, 12, 0, 13, 4, 3, 0),
+    gsSP2Triangles(14, 15, 16, 0, 17, 18, 19, 0),
+    gsSP2Triangles(20, 18, 17, 0, 2, 21, 0, 0),
+    gsSP2Triangles(22, 23, 24, 0, 25, 23, 22, 0),
+    gsSP2Triangles(26, 27, 28, 0, 29, 27, 26, 0),
+    gsSPVertex(&sElegyShellHumanVtx[313], 24, 0),
+    gsSP2Triangles(0, 1, 2, 0, 3, 1, 0, 0),
+    gsSP2Triangles(4, 5, 6, 0, 7, 8, 9, 0),
+    gsSP2Triangles(10, 11, 12, 0, 13, 10, 12, 0),
+    gsSP2Triangles(14, 8, 7, 0, 15, 16, 17, 0),
+    gsSP2Triangles(18, 19, 20, 0, 21, 22, 23, 0),
+    gsSPDisplayList(0x0C000000),
+    gsDPSetTextureLUT(G_TT_NONE),
+    gsDPLoadTextureBlock(gElegyShellHumanPlatformTex, G_IM_FMT_RGBA, G_IM_SIZ_16b, 8, 16, 0, G_TX_NOMIRROR | G_TX_WRAP,
+                         G_TX_NOMIRROR | G_TX_WRAP, 3, 4, G_TX_NOLOD, G_TX_NOLOD),
+    gsDPSetPrimColor(0, 0x80, 255, 255, 255, 255),
+    gsSPVertex(&sElegyShellHumanVtx[337], 23, 0),
+    gsSP2Triangles(0, 1, 2, 0, 0, 2, 3, 0),
+    gsSP2Triangles(4, 5, 6, 0, 4, 6, 7, 0),
+    gsSP2Triangles(3, 2, 8, 0, 3, 8, 9, 0),
+    gsSP2Triangles(7, 6, 10, 0, 7, 10, 11, 0),
+    gsSP2Triangles(12, 13, 3, 0, 4, 14, 12, 0),
+    gsSP2Triangles(12, 14, 11, 0, 3, 15, 12, 0),
+    gsSP2Triangles(16, 17, 18, 0, 19, 17, 20, 0),
+    gsSP2Triangles(18, 17, 21, 0, 20, 17, 22, 0),
+    gsSPEndDisplayList(),
+};
+
+// Shells for each of Link's different forms
+// (Playing elegy as Fierce Deity puts down a human shell)
+static Gfx* sShellDLists[] = {
+    gElegyShellHumanDL, // gElegyShellGoronDL, gElegyShellZoraDL, gElegyShellDekuDL, gElegyShellHumanDL,
+};
+
+void EnTorch2_Init(Actor* thisx, GameState_Play* play) {
+    EnTorch2* this = THIS;
+
+    Actor_ProcessInitChain(&this->base, sInitChain);
+    Collider_InitCylinder(play, &this->collider);
+    Collider_SetCylinder(play, &this->collider, &this->base, &sCylinderInit);
+
+    if (this->base.variable != TORCH2_PARAM_DEKU) {
+        this->base.flags |= (1 << 26); // ACTOR_FLAG_CAN_PRESS_SWITCH;
+
+        // All OoT Elegy Statues are human, for now
+        // Also no switches require a "heavy" actor, and rusty switches work completely differently
+        // if (this->base.variable == TORCH2_PARAM_GORON) {
+        //     this->base.flags |= ACTOR_FLAG_CAN_PRESS_HEAVY_SWITCH;
+        // }
+    }
+    this->framesUntilNextState = 20;
+}
+
+void EnTorch2_Destroy(Actor* thisx, GameState_Play* play) {
+    EnTorch2* this = THIS;
+
+    Collider_DestroyCylinder(play, &this->collider);
+
+    gElegyShells[this->base.variable] = NULL;
+}
+
+#define fabsf(f) __builtin_fabsf((f32)(f))
```

Comment:
This should:

1) Be in a header
2) Not cast as implicit type convesrions can give useful warnings